src/ol/render/canvas/canvasreplay.js

```diff
@@ -258,6 +265,10 @@ ol.render.canvas.Replay.prototype.replay_ = function(
   var localTransform = this.tmpLocalTransform_;
   var localTransformInv = this.tmpLocalTransformInv_;
   var prevX, prevY, roundX, roundY;
+  var pendingFill = 0;
+  var pendingStroke = 0;
+  var batchSize =
+      this.instructions != instructions || this.overlaps ? 0 : 200;
```

Comment:
Curious about the `200` here.  Did you find that performance is degraded if the batch size gets too big?  It might be nice to document the justification for this default on its own line.